pkg/collector/corechecks/net/networkv2/network_windows.go

```diff
@@ -0,0 +1,352 @@
+	return net.ProtoCounters(protocols)
+}
+
+func (n defaultNetworkStats) Connections(kind string) ([]net.ConnectionStat, error) {
+	return net.Connections(kind)
+}
+
+func (n defaultNetworkStats) TcpStats(kind string) (*mibTcpStats, error) {
+	return getTcpStats(kind)
+}
+
+// Run executes the check
+func (c *NetworkCheck) Run() error {
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
+	}
+
+	// _cx_state_psutil
+	if c.config.instance.CollectConnectionState {
+		for _, protocol := range []string{"udp4", "udp6", "tcp4", "tcp6"} {
+			connectionsStats, err := c.net.Connections(protocol)
+			if err != nil {
+				return err
+			}
+			submitConnectionsMetrics(sender, protocol, connectionsStats)
+		}
+	}
+
+	// _cx_counters_psutil
+	ioByInterface, err := c.net.IOCounters(true)
+	if err != nil {
+		return err
+	}
+
+	for _, interfaceIO := range ioByInterface {
+		if !c.isDeviceExcluded(interfaceIO.Name) {
+			submitInterfaceMetrics(sender, interfaceIO)
+		}
+	}
+
+	// _tcp_stats
+	err = c.submitTcpStats(sender)
+	if err != nil {
+		return err
+	}
+
+	sender.Commit()
+	return nil
+}
+
+func submitConnectionsMetrics(sender sender.Sender, protocolName string, connectionsStats []net.ConnectionStat) {
+	metricCount := map[string]float64{}
+	var stateMetricSuffixMapping map[string]string
+	switch protocolName {
+	case "udp4", "udp6":
+		stateMetricSuffixMapping = udpStateMetricsSuffixMapping
+	case "tcp4", "tcp6":
+		stateMetricSuffixMapping = tcpStateMetricsSuffixMapping
+	}
+	for _, suffix := range stateMetricSuffixMapping {
+		metricCount[suffix] = 0
+	}
+
+	for _, connectionStats := range connectionsStats {
+		metricCount[stateMetricSuffixMapping[connectionStats.Status]]++
+	}
+
+	for suffix, count := range metricCount {
+		sender.Gauge(fmt.Sprintf("system.net.%s.%s", protocolName, suffix), count, "", nil)
+	}
+}
+
+const (
+	AF_INET  = 2
+	AF_INET6 = 23
+)
+
+// TCPSTATS DWORD mappings
+// https://learn.microsoft.com/en-us/windows/win32/api/tcpmib/ns-tcpmib-mib_tcpstats_lh
+type mibTcpStats struct {
+	DwRtoAlgorithm uint32
+	DwRtoMin       uint32
+	DwRtoMax       uint32
+	DwMaxConn      uint32
+	DwActiveOpens  uint32
+	DwPassiveOpens uint32
+	DwAttemptFails uint32
+	DwEstabResets  uint32
+	DwCurrEstab    uint32
+	DwInSegs       uint32
+	DwOutSegs      uint32
+	DwRetransSegs  uint32
+	DwInErrs       uint32
+	DwOutRsts      uint32
+	DwNumConns     uint32
+}
+
+var (
+	iphlpapi               = windows.NewLazySystemDLL("iphlpapi.dll")
+	procGetTcpStatisticsEx = iphlpapi.NewProc("GetTcpStatisticsEx")
+)
+
+// gopsutil does not call GetTcpStatisticsEx so we need to do so manually for these stats
+func getTcpStats(protocolName string) (*mibTcpStats, error) {
+	var inet uint
+	switch protocolName {
+	case "tcp4":
+		inet = AF_INET
+	case "tcp6":
+		inet = AF_INET6
+	}
+	tcpStats := &mibTcpStats{}
+	r0, _, _ := syscall.SyscallN(procGetTcpStatisticsEx.Addr(), uintptr(unsafe.Pointer(tcpStats)), uintptr(inet), 0)
```

Comment:
```suggestion
	r0, _, _ := procGetTcpStatisticsEx.Call(uintptr(unsafe.Pointer(tcpStats)), uintptr(inet), 0)
```

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/37240/commits/a8f06404638e4898f0962b431c7e75a65bd21ed9#diff-f8bfa24f1e3f0888eca542771acc5c5a2fa394dacf5e279c0c0955c8d97fdb41R212